### ISSUE TITLE
Node doesn't stop when running from Babun

### ISSUE THREAD

**Issue by TravelingTechGuy** [babun/babun#391]
When I run a Node server, it does not stop when I hit Ctrl+C. Instead, it keeps running in the background. Nor does it appear when I issue `ps`. I have to open Task Manager, locate `node.exe` and kill it.
The side effect is that you cannot run node again, since the port is taken - until you kill it.

Babun 1.2.0, Node 0.12.5

**Comment by TravelingTechGuy** [comment]
It's 2 months later. I'm now running Node 4.0, and I can no longer use Babun as my dev console, since this issue still occurs. Any time you start a Node process, 2 NodeJS instances are started. When you stop yours, one always hangs around. And unless you kill it in task manager, you cannot restart your own.

To me, this ia bug. I do not know why it was labeled Feedback.

**Comment by PSeitz** [comment]
+1 Same here. Pretty unusable for node development.

**Comment by 2c2c** [comment]
looking around this seems like a lifelong issue since the dawn of cygwin, right? there are hacky solutions out there, but there should really be something builtin.

**Comment by TravelingTechGuy** [comment]
Really? I'm using Cygwin in ConEMU for my Node development, and have not run against that issue. `Ctrl+C` kills the process 100% of the times. I'm not sure what in the way Babun uses Cygwin may cause this, but until we have a good workaround, it's quite a pain.

**Comment by julestruong** [comment]
Still the problem on 02/2016

**Comment by 2c2c** [comment]
If you look up sigint cygwin you will see this is some issue with running
native windows processes through cygwin terminal

**Comment by TravelingTechGuy** [comment]
@2c2c I run Node in my Cygwin window all the time (under ConEmu), and I've never run into this before. Something else is tying up the process.

**Comment by ingshtrom** [comment]
Yeah, this stinks. I was really hoping to use Babun but I really can't. We run our dev server with `piping` for HMR and that spawns a child process... so unless you only need to run the app once, you are stuck ><

**Comment by kenberkeley** [comment]
the problem still exists, 2016/04/13

**Comment by johniak** [comment]
still exist

**Comment by Fender123** [comment]
same here

**Comment by rettgerst** [comment]
I'm not using babun (mintty) but I have the same issue. I guess it's more of a problem with cygwin?

**Comment by kenberkeley** [comment]
Cygwin is fine, so it's most likely a zsh problem?

**Comment by 2c2c** [comment]
It's zsh there are very old mailing list discussions if you Google around

**Comment by nonoroazoro** [comment]
not solved yet

**Comment by Alber70g** [comment]
Still experiencing problems

**Comment by ChiralAlchemist** [comment]
Same issue this kills the node server (not a permanent solution)

this will show you your node process id
netstat -a -o | grep <*ur node port #*>

this shuts down node server
taskkill /F /PID** <*ur Process ID #*>

**Comment by Castrovalva** [comment]
Why does this issue has the label 'feedback'? This is a horrible, I would love to use Babun, but when this is not fixed it is not usefull to use it...

**Comment by nonoroazoro** [comment]
@Castrovalva I've met this problem before. It seems that you can solve this problem by updating cygwin (run **/.babun/update.bat**).

**Comment by Castrovalva** [comment]
@nonoroazoro Thanks it works now 👍

**Comment by jawilson** [comment]
I'm on the latest babun/cygwin. Still an Issue.
```
{ ~ }  » babun update                                                                                                                                                                                                                    
Executing babun update
  upstream  [release]
  installed [1.2.0]
  newest    [1.2.0]
Babun is up to date
Checking Cygwin version:
  installed [1.7.35]
  newest    [1.7.35]
Cygwin is up to date
```

**Comment by kevinhaas** [comment]
Same here, fully updated Babun/Cygwin using ohmyzsh.

**Comment by mateodelnorte** [comment]
+1 having this problem. I went the stop-gap route and created this https://github.com/mateodelnorte/kill-all-node-on-windows which I run before I start/restart my processes. 

Note, running node-supervisor does not seem to leak orphaned node processes when it recycles them on detecting file changes. However, if you `ctrl-c` the process you'll be left with 2 processes hung. And if you do something modern like have the following task defined in a package.json script `cross-env concurrently \"supervisor myapp\" \"supervisor myapi\"', you'll be left with four hanging processes. 

This is pretty darn bad. :-\ But, I fear it's a cygwin problem and deeper than babun.

**Comment by BennettStaley** [comment]
Can confirm, issue persists. Quite annoying.

**Comment by Nek** [comment]
Same here. I also run tomcat from Babun and obviously CTRL+C doesn't work for it too.
Have to use a workaround:
```
powershell kill -n node
powershell kill -n java
```

**Comment by ralberts** [comment]
I am running into the same issue :-(

**Comment by Nek** [comment]
I've just switched to Chocolatey and Git Bash combo. Works just fine.

**Comment by TopGrd** [comment]
I have the same issuee, I can't use `ctrl + c` to stop a node process. for example, i use `npm start` to run a express project, then i press `ctrl + c`, the node process was not stop, i must close the bash.  
I use cmder + babun + oh my zsh on Windows 10.

**Comment by romaindso** [comment]
Same problem here with Babun + Oh My Zsh on Windows 10. The node process isn't kill when i press `ctrl + c`

**Comment by doublnt** [comment]
Still Exist

**Comment by Suzal3579** [comment]
About 2 years now, and the issue still is alive. 👎

**Comment by chrisjbreisch** [comment]
@Suzal3579, this project appears to be nearly, if not completely, dead. See [#774](https://github.com/babun/babun/issues/774). Perhaps it can be resurrected with some help. See [#712](https://github.com/babun/babun/issues/712).

Personally, I've switched back to vanilla Cygwin. I get 64-bit, somewhat frequent updates, and better reliability. Perhaps Babun is easier to set up, but I've already gone through that pain, and I like my Cygwin environment the way that I've set it up. So I don't feel that I ever really gained anything with Babun.

I don't think that has to be the case. I think that Babun offers some intriguing possibilities. These possibilities remain unfulfilled at the moment, however.

**Comment by planesweep** [comment]
The issue is still exists and is not solved in babun 1.2.0 

Babun is the shell you will hate!!!

**Comment by OogieBoogieInJSON** [comment]
Cygwin, Babun, zsh, whatever, they send a different kill signal that is not recognized by Windows OS.

I'm using kill aliases, nodejs as example:

```
kn='taskkill /F /IM node.exe'
```

After ctrl+c, just type kn and all node instances are killed. This might be an issue if you have multiple running and don't want to end them all, you will need the PID then.

**Comment by edunad** [comment]
Any updates?

**Comment by irbrownie** [comment]
Can confirm, still a problem.

**Comment by zubair1024** [comment]
Wow, this has been long pending... Sigh..

**Comment by Albuseven** [comment]
still exsit. time to change a shell

**Comment by eXistenZx** [comment]
It seems Babun team doesn't pay attention to so much complaints for so long time..
For my part, I move to another shell..
Good luck guys..

**Comment by prameshbajra** [comment]
@eXistenZx I did the same. But, with a twist. 
`Moved to LINUX` :laughing:

**Comment by zubair1024** [comment]
> It seems Babun team doesn't pay attention to so much complaints for so long time..
> For my part, I move to another shell..
> Good luck guys..

Which shell did you move to? Something based on Zsh?

**Comment by chrisjbreisch** [comment]
> I'm now on MobaXterm, it's perfect.. The free version is more than enough for me..

Interesting. Had not heard of that. It appears to be a graphical wrapper around Cygwin? Is it 32 or 64 bit? How do they stay up-to-date with the underlying Cygwin tools? Or do they?

**Comment by wenwuwu** [comment]
still exist, unbelievable.

**Comment by SpartakK** [comment]
Stupid, I cannot kill git in Babun Ctrl+C does not work

**Comment by SpartakK** [comment]
Found, if you want to stop git in working dir in Babun, just remove it
{somedir} master  >> rm -rf .git
{somedir}  >>
